blogs/2022-06-28-tips-kafka-streams-distributed.md

```diff
@@ -115,7 +121,7 @@ Our first assumption was `all()` returns an object (Flow in our case), as the AP
 - Deserialize the data from RocksDB that is stored as byte and map it to concrete Java POJO
 
 
-So each time we call `all()` on the method, all values are deserialized, which can lead to high CPU usage and latency on your stream. We are talking about all flows revision on our cluster for our use case. 2.5K flows (last revision), but we don't see people creating a lot of revisions (100K). Imagine 100K `byte[]` to deserialize to POJO for every call (mostly all the execution end).
+So each time we call `all()` on the method, all values are deserialized, which can lead to high CPU usage and latency on your stream. We are talking about all flow revisions on our cluster. 2.5K flows (last revision), but we don't see people creating a lot of revisions (100K). Imagine 100K `byte[]` to deserialize to POJO for every call (mostly all the execution end).
```

Comment:
- Do the revision mean the built-in source control for the flows?
- I couldn't understand this section: 2.5K flows (last revision), but we don't see people creating a lot of revisions (100K). Imagine 100K `byte[]` to deserialize to POJO for every call (mostly all the execution end).

Reply:
- internal revision inside kestra, we use simple integer increment and keep all revision: so yes ;) 
- It's mostly some numbers example, in our case, we detect huge cpu usage with only 2500 flows, and I imagine large cluster with 100 000 flows, that sound really possible the impact on the CP, this one is mostly hidden from the developer perspective